### PR TITLE
Use GitHub Actions for CI

### PR DESCRIPTION
**Pull request by orien**
### Context

[travis-ci.org](https://www.travis-ci.org/) is dead and the current CI pipeline is not working.

### Change

To ensure compatibility with various versions of Ruby, I propose using GitHub Actions for CI. The proposed workflow is [passing on my fork](https://github.com/orien/persistent-cache/actions/runs/4096817294), and will be added to this repo with the merge of this pull request.

### CODE REVIEW

**Comment by orien** [comment]
@JoshuaSmeda What do you think of this idea?

**Comment by JoshuaSmeda** [comment]
Test coverage step seems to be failing on all the matrix's too:

Do you perhaps have a fix for this too?

```
[Coveralls] Submitting to https://coveralls.io/api/v1
Coveralls encountered an exception:
JSON::ParserError
785: unexpected token at '<html>
<head><title>400 Bad Request</title></head>
<body>
<center><h1>400 Bad Request</h1></center>
<hr><center>nginx</center>
</body>
</html>
'
/home/runner/work/persistent-cache/persistent-cache/vendor/bundle/ruby/2.6.0/gems/json-2.2.0/lib/json/common.rb:156:in `parse'
/home/runner/work/persistent-cache/persistent-cache/vendor/bundle/ruby/2.6.0/gems/json-2.2.0/lib/json/common.rb:156:in `parse'
/home/runner/work/persistent-cache/persistent-cache/vendor/bundle/ruby/2.6.0/gems/json-2.2.0/lib/json/common.rb:335:in `load'
/home/runner/work/persistent-cache/persistent-cache/vendor/bundle/ruby/2.6.0/gems/coveralls-0.8.23/lib/coveralls/api.rb:31:in `post_json'
/home/runner/work/persistent-cache/persistent-cache/vendor/bundle/ruby/2.6.0/gems/coveralls-0.8.23/lib/coveralls/simplecov.rb:64:in `format'
/home/runner/work/persistent-cache/persistent-cache/vendor/bundle/ruby/2.6.0/gems/simplecov-0.16.1/lib/simplecov/result.rb:48:in `format!'
/home/runner/work/persistent-cache/persistent-cache/vendor/bundle/ruby/2.6.0/gems/simplecov-0.16.1/lib/simplecov/configuration.rb:[182](https://github.com/orien/persistent-cache/actions/runs/4096817294/jobs/7064732725#step:4:183):in `block in at_exit'
/home/runner/work/persistent-cache/persistent-cache/vendor/bundle/ruby/2.6.0/gems/simplecov-0.16.1/lib/simplecov.rb:200:in `run_exit_tasks!'
/home/runner/work/persistent-cache/persistent-cache/vendor/bundle/ruby/2.6.0/gems/simplecov-0.16.1/lib/simplecov/defaults.rb:27:in `block in <top (required)>'
```

**Comment by orien** [comment]
@JoshuaSmeda I've pushed up a fix for that. It required updating [simplecov](https://rubygems.org/gems/simplecov) to a new version, with requires Ruby 2.5 (to run the test suite).

Passing build here: https://github.com/orien/persistent-cache/actions/runs/4098643081
Coverage reported here: https://coveralls.io/github/orien/persistent-cache

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+name: Tests
+on: [ push, pull_request ]
+jobs:
+  test:
+    name: test (Ruby ${{ matrix.ruby }})
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        ruby: [ '2.5', '2.6', '2.7' ]
```
Comment (JoshuaSmeda) [inline .github/workflows/test.yml:11]: Regressed here, can we add in Ruby 3?

**Review by JoshuaSmeda — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (orien) [reply .github/workflows/test.yml:11]: No, the gemspec only allows Ruby 2. 

https://github.com/xneelo/persistent-cache/blob/0686104b9110faee96f6ab43e9bc4a478de7cec3/persistent-cache.gemspec#L19

Changing this is out of scope for this PR.

Comment (orien) [reply .github/workflows/test.yml:11]: @JoshuaSmeda I've proposed this change in #10.

**Pull request merged**